scout/adapter/mongo/query.py

```diff
@@ -324,17 +338,56 @@ def coordinate_filter(self, query, mongo_query):
             mongo_query(dict): returned object contains coordinate filters
 
         """
-        LOG.debug('Adding genomic coordinates to the query')
-        chromosome = query['chrom']
-        mongo_query['chromosome'] = chromosome
-
-        if (query.get('start') and query.get('end')):
-            mongo_query['position'] = {'$lte': int(query['end'])}
-            mongo_query['end'] = {'$gte': int(query['start'])}
+        mongo_query['position'] = {'$lte': int(query['end'])}
+        mongo_query['end'] = {'$gte': int(query['start'])}
 
         return mongo_query
 
 
+    def sv_coordinate_filter(self, query, mongo_query):
+        """ Adds genomic coordinated-related filters to the query object
+            This method is called to buid coordinate query for sv variants
+
+        Args:
+            query(dict): a dictionary of query filters specified by the users
+            mongo_query(dict): the query that is going to be submitted to the database
+
+        Returns:
+            coordinate_query(dict): returned object contains coordinate filters for sv variant
+
+        """
+        # Query for overlapping intervals. Taking into account these cases:
+        #1
+        # filter                 xxxxxxxxx
+        # Variant           xxxxxxxx
+
+        #2
+        # filter                 xxxxxxxxx
+        # Variant                    xxxxxxxx
+
+        #3
+        # filter                 xxxxxxxxx
+        # Variant                   xx
+
+        #4
+        # filter                 xxxxxxxxx
+        # Variant             xxxxxxxxxxxxxx
+        coordinate_query = {
+            "$or": [
+                { "end": { "$gte": int(query['start']), "$lte": int(query['end']) }}, #1
```

Comment:
Hm, you still need to check chromosome as well. BNDs appear with different start and end chromosomes.

Reply:
Right! On it!